XiEditor/Storage/SettingsProvider.swift

```diff
@@ -0,0 +1,36 @@
+//
+//  SettingsStore.swift
+//  XiEditor
+//
+//  Created by Alex Demishkevych on 10/13/18.
+//  Copyright © 2018 Raph Levien. All rights reserved.
+//
+
+import Foundation
+
+protocol SettingsProviding {
+    func fetchingRecentlyUsedLanguages() -> [String]
+    func saveRecentlyUsed(languages: [String])
+}
+
+class SettingsProvider: SettingsProviding {
+
+    enum Keys: String {
+        case recentlyUsedLanguages
+    }
+
+    private let keyValueStorage: KeyValueStorage
```

Comment:
Hi @jansol, thanks for the comment! I have introduced `KeyValueStorage ` protocol in order to have ability easily switch implementations without touching client code.

First benefit of such approach is that It allows to choose whatever key-value storage is needed (UserDefaults, Dictionary etc.). It is a rare case to be honest, but still.
Second benefit goes when it is needed to mock and stub such storages in tests. The main disadvantage working with `UserDefaults` is that `UserDefaults` persists the data between tests launch. From my experience it is very easy to forget to remove user defaults suite or  some value and eventually find out that not related test cases are falling because of not properly cleaned up `UserDefaults`. So, it is very easy to mess it up.

Having such protocol allows to create mock `DictionaryStorage` and use it as storage in tests.

What do you think? Does it make sense?

Reply:
It makes sense for mocking, yes. But I didn't see any case where this *actually* is anything other than `DictionaryStorage` (and no other implementations either), hence my suspicions.

Reply:
@jansol yes, I agree. So, do we keep the protocol? I think it worth keeping it, at least for mocking purposes. I am open to any other suggestions if any.